OsmAnd/src/net/osmand/plus/plugins/CustomOsmandPlugin.java

```diff
@@ -83,6 +93,77 @@ public CustomOsmandPlugin(@NonNull OsmandApplication app, @NonNull JSONObject js
 		readAdditionalDataFromJson(json);
 		readDependentFilesFromJson(json);
 		loadResources();
+		updateCustomRegionsState();
+	}
+
+	private void updateCustomRegionsState() {
+		ArrayList<Pair<File, File>> filesToCopy = new ArrayList();
+		for (WorldRegion region : getFlatCustomRegions()) {
+			if (region instanceof CustomRegion) {
+				CustomRegion customRegion = (CustomRegion) region;
+				List<IndexItem> indexItems = customRegion.loadIndexItems();
+				for (IndexItem item : indexItems) {
+					if (item.isHidden()) {
+						item.setIsHidden(false);
+						File nonHiddenFile = item.getTargetFile(app);
+						if (nonHiddenFile.exists()) {
+							item.setIsHidden(true);
```

Comment:
isHidden->setIsHidden(false)->.setIsHidden(true) looks strange for getting hidden and nonHidden target files